src/org/thoughtcrime/securesms/ProfileGalleryFragment.java

```diff
@@ -184,39 +188,84 @@ public void onMediaLongClicked(DcMsg mediaRecord) {
     }
   }
 
-  @SuppressLint("StaticFieldLeak")
-  private void handleDeleteMedia(@NonNull Collection<DcMsg> mediaRecords) {
-    int recordCount       = mediaRecords.size();
-    Resources res         = getContext().getResources();
-    String confirmMessage = res.getQuantityString(R.plurals.ask_delete_messages,
-                                                  recordCount,
-                                                  recordCount);
-
-    AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
-    builder.setMessage(confirmMessage);
-    builder.setCancelable(true);
-
-    builder.setPositiveButton(R.string.delete, (dialogInterface, i) -> {
-      new ProgressDialogAsyncTask<DcMsg, Void, Void>(getContext(),
-                                                                         R.string.one_moment,
-                                                                         R.string.one_moment)
-      {
-        @Override
-        protected Void doInBackground(DcMsg... records) {
-          if (records == null || records.length == 0) {
-            return null;
-          }
-
-          for (DcMsg record : records) {
-            dcContext.deleteMsgs(new int[]{record.getId()});
-          }
-          return null;
-        }
-
-      }.execute(mediaRecords.toArray(new DcMsg[mediaRecords.size()]));
+  private void handleDisplayDetails(DcMsg dcMsg) {
+    String info_str = dcContext.getMsgInfo(dcMsg.getId());
+    AlertDialog d = new AlertDialog.Builder(getActivity())
+            .setMessage(info_str)
+            .setPositiveButton(android.R.string.ok, null)
+            .create();
+    d.show();
+    try {
+      //noinspection ConstantConditions
+      Linkify.addLinks((TextView) d.findViewById(android.R.id.message), Linkify.WEB_URLS);
+    } catch(NullPointerException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void handleDeleteMedia(final Set<DcMsg> messageRecords) {
+    int messagesCount = messageRecords.size();
+
+    new AlertDialog.Builder(getActivity())
+            .setMessage(getActivity().getResources().getQuantityString(R.plurals.ask_delete_messages, messagesCount, messagesCount))
+            .setCancelable(true)
+            .setPositiveButton(R.string.delete, (dialog, which) -> {
+                int[] ids = DcMsg.msgSetToIds(messageRecords);
+                dcContext.deleteMsgs(ids);
```

Comment:
You moved the deletion here to the main thread. Do you know if the deletion happens async in the core?

Reply:
I basically tried to be consistent because it was done in one approach here and another in ConversationFragment.java since I actually don't fully understand what I am doing :) I decided to take the simple one, if this is wrong please consider fixing it yourself here and in ConversationFragment.java, help is appreciated

Reply:
The deletion does not happen async in the core. For every deleted message, there is one read access (loading the message) and two write accesses (trashing the message locally and saving a job to delete it on the server) to the SQL db.

I just deleted 170 messages, and it went fine; according to the log, it took 1.04s.

On the other hand - I'll just make a commit that copy-pastes uses the async approach, just because it's done so easily.

Reply:
Actually actually, let's leave things as they are I'd say. The async approach is waaaaay slower because it calls deleteMsgs() for each message on its own, instead of all at once. And no one complained until now (we always used the synchronous approach in the chat activity), so let's focus on things people complain about :)